lib/srv/uacc/uacc.h

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef UACC_C
+#define UACC_C
+
+#include <unistd.h>
+#include <stdint.h>
+#include <string.h>
+#include <utmp.h>
+#include <errno.h>
+#include <stdbool.h>
+
+int UACC_UTMP_MISSING_PERMISSIONS = 1;
+int UACC_UTMP_WRITE_ERROR = 2;
+int UACC_UTMP_READ_ERROR = 3;
+int UACC_UTMP_FAILED_OPEN = 4;
+int UACC_UTMP_ENTRY_DOES_NOT_EXIST = 5;
+
+// I opted to do things with setutent/pututline etc manually instead of using the login/logout BSD functions due to
+// running into some weird behaviour. Upon asking on IRC I was told to avoid these with a 10 foot pole and stick to this.
```

Comment:
Can you document what it was? Will help future developers when they run into the same issue to understand why you made the choices you did.

Reply:
Sure!